File: cookbooks/HuggingFace/python/ask-mistral.py
import asyncio

from aiconfig.model_parser import InferenceOptions

from aiconfig import AIConfigRuntime


async def main():
    config = AIConfigRuntime.load("../Mistral-aiconfig.json")

    def stream_callback(data, accumulated_message, index):
        print(data, end="")

    inference_options = InferenceOptions(stream_callback=stream_callback)
    await config.run("prompt1", options=inference_options)


if __name__ == "__main__":
    asyncio.run(main())

File: cookbooks/HuggingFace/typescript/ask-mistral.ts
import {
  Prompt,
  Output,
  PromptInput,
  ModelMetadata,
  ExecuteResult,
  AIConfigRuntime,
  InferenceOptions,
  InferenceCallbackHandlers,
  ModelParserRegistry,
} from "aiconfig";

import { HuggingFaceTextGenerationModelParser } from "aiconfig-extension-huggingface-textgeneration";

async function main() {
  const textGenerationModelParser = new HuggingFaceTextGenerationModelParser();

  AIConfigRuntime.registerModelParser(
    new HuggingFaceTextGenerationModelParser()
  );

  const config = AIConfigRuntime.load("../Mistral-aiconfig.json");

  function consoleLogCallback(
    data: string,
    accumulatedMessage: string,
    index: number
  ) {
    process.stdout.write(data);
  }

  const callbacks: InferenceCallbackHandlers = {
    streamCallback: consoleLogCallback,
  };

  const inferenceOptions: InferenceOptions = { callbacks, stream: true };

  await config.run("prompt1", undefined, inferenceOptions);

  config.save(undefined, {serializeOutputs: true});
}

main();
